### ISSUE TITLE
Synapse start error: NameError: global name 'sys' is not defined

### ISSUE THREAD

**Issue by maxidor** [matrix-org/synapse#2503]
### Description
synapse doesn't start anymore after update to 0.23 using Deb packages on Linux Mint 18
```
twisted - 131 - ERROR - - Traceback (most recent call last):
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
twisted - 131 - ERROR - -     "__main__", fname, loader, pkg_name)
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
twisted - 131 - ERROR - -     exec code in run_globals
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 448, in <module>
twisted - 131 - ERROR - -     main()
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 443, in main
twisted - 131 - ERROR - -     hs = setup(sys.argv[1:])
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 307, in setup
twisted - 131 - ERROR - -     hs.run_startup_checks(db_conn, database_engine)
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/dist-packages/synapse/app/homeserver.py", line 229, in run_startup_checks
twisted - 131 - ERROR - -     % (self.hostname,)
twisted - 131 - ERROR - -   File "/usr/lib/python2.7/dist-packages/synapse/app/_base.py", line 113, in quit_with_error
twisted - 131 - ERROR - -     sys.stderr.write("*" * line_length + '\n')
twisted - 131 - ERROR - - NameError: global name 'sys' is not defined
```

### Steps to reproduce
- Update synapse from pre 0.23 to 0.23 using Matrix deb repo
- Synapse fails to start

**Comment by richvdh** [comment]
It looks like you're running 3fed5bb2, which is between 0.23 and 0.23.1: this was fixed before 0.23.1 (by ea87cb1b)

So the solution here is to use a released version...

**Comment by rth9** [comment]
The line missing is import sys from the commit to 0.23.1. I had the same issue using the repo as well.

Edit the following and add import.sys where it says from ea87cb1:
/usr/lib/python2.7/dist-packages/synapse/app/_base.py

**Comment by erikjohnston** [comment]
This was fixed in debian package 0.23.1-1

**Comment by maxidor** [comment]
It wasn't:
```
max@Max-Desktop-Linux ~ $ apt-get upgrade
[...]
The following packages will be upgraded:
  git git-man matrix-synapse peek
[...]
Get:2 https://matrix.org/packages/debian xenial/main amd64 matrix-synapse all 0.23.1-1 [485 kB]
[...]
Unpacking matrix-synapse (0.23.1-1) over (0.23.0-1) ...
[...]
Setting up matrix-synapse (0.23.1-1) ...
[...]
max@Max-Desktop-Linux ~ $ cd /etc/matrix-synapse/conf.d
max@Max-Desktop-Linux /etc/matrix-synapse/conf.d $ cat server_name.yaml 
server_name: localhost
```
